components/script/dom/console.rs

```diff
@@ -30,18 +30,81 @@ const MAX_LOG_CHILDREN: usize = 15;
 /// <https://developer.mozilla.org/en-US/docs/Web/API/Console>
 pub struct Console;
 
+#[allow(unsafe_code)]
+fn get_js_stack(cx: *mut jsapi::JSContext) -> Vec<StackFrame> {
+    const MAX_FRAME_COUNT: Option<u32> = Some(128);
+
+    let mut frames = vec![];
+    rooted!(in(cx) let mut handle =  ptr::null_mut());
+    let captured_js_stack = unsafe { CapturedJSStack::new(cx, handle, MAX_FRAME_COUNT) };
```

Comment:
Can we use the capture_stack macro here instead? https://github.com/servo/mozjs/blob/main/mozjs/tests/capture_stack.rs#L66

Reply:
In theory, yes, but `capture_stack` requires an `unsafe` block. And because we need the captured stack through the whole function, that would require the whole function body to be `unsafe`. I usually prefer to keep unsafe blocks as small as possible :/